rustls/src/vecbuf.rs

```diff
@@ -117,13 +115,83 @@ impl ChunkVecBuffer {
             return Ok(0);
         }
 
-        let used = wr.write_vectored(
-            &self
-                .chunks
-                .iter()
-                .map(|ch| io::IoSlice::new(ch))
-                .collect::<Vec<io::IoSlice>>(),
-        )?;
+        // XXX: `rustls`'s MSRV is 1.40.0, where `IoSlice` doesn't implement
+        // `Copy`. When the MSRV is Rust 1.44.0, this could be the much more
```

Comment:
We should update the MSRV required for the current `main` branch. There have been some really important improvements, especially related to arrays and min_const_generics and friends. I think we should increase the MSRV at least to 1.44.0 and beyond as necessary, to allow us to do things like this with better code.

Reply:
I proposed increasing the MSRV to 1.47.0 in PR #463.